src/Utils/Utils.php

```diff
@@ -223,4 +223,26 @@ public static function isEmpty($value): bool
 
         return false;
     }
+
+    private static function getPage(Request $request): int
+    {
+        $page = $request->query->getInt('p', 1);
+
+        if ($request->isMethod(Request::METHOD_POST)) {
+            $page = $request->request->getInt('p', $page);
+        }
+
+        return $page <= 0 ? 1 : $page;
+    }
+
+    public static function getOffset(Request $request, ?int $limit = null)
+    {
+        if (!$limit) {
+            $limit = Pagination::DEFAULT_LIMIT;
+        }
+
+        $page = self::getPage($request);
+
+        return ($page - 1) * $limit;
+    }
```

Comment:
These are no utility functions, because other than in search, this is basically useless. Put these methods into a class that represents their usage.